### PR TITLE
Improvements based on improvement proposals

### PR DESCRIPTION
**Pull request by awjh-ibm**
PR based on improvements specified in https://github.com/hyperledger/composer/issues/3130﻿

# Changes made
## General 
- Changed vehicle-lifecycle naming to vehicle-manufacture
- Configured applications for running in iframes

## Installer
- Updated references to vehicle-lifecycle-network business network to be vehicle-manufacture-network
- Updated flows.json to remove previous Node-RED implementation and replaced it with a solution that listens for an UpdateOrderStatusEvent and sends and email if status is updated to delivery, the user paul has an email set and the user of the demo has configured the send email node. 
- Added install for using fabric v1.1 which installs against composer <span>@</span>next
- Install now listens for an environment variable of DEMO_FABRIC_VERSION if unset it defaults to fabric 1.0 and composer <span>@</span>latest, if set to hlfv11 it uses fabric 1.1 and composer <span>@</span>next

## Car builder
- Updated config to include reference to the REST server details
- Updated app.js to return the JSON config unless the environment variable REST_SERVER_CONFIG is set with the REST server details.
- Updated building screen to show tick when selecting trim and extra options
- Updated building screen to prevent submission without filling in trim, colour and interior details for the car
- Updated building screen to submit PlaceOrder transaction to REST server
- Updated PlaceOrder request format to reflect changes in the business network
- Updated status screen to listen to websockets coming from the REST server

## Manufacturing
- Updated config to include reference to the REST server details
- Added app.set to server/app.js that stores value from config file or if set the environment variable REST_SERVER_CONFIG
- Updated websocket in server/app.js to prevent app death on disconnection from remote and set it to use the config set so that it listens to the REST server for events
- Removed vehicles and transactions endpoints.
- Added orders endpoint to retrieve PlaceOrder transactions using the REST server and combine with retrieved UpdateOrderStatus transactions to handle that order assets no longer store an array of the updates placed on them
- Added updateOrderStatus endpoint to submit UpdateOrderStatus transactions via the REST server
- Updated dashboard controller loading of statusUpdates to handle the way they are returned by the orders endpoint
- Updated dashboard controller to use a single websocket for all events
- Updated updating of order status' in dashboard controller to use updateOrderStatus endpoint

## VDA
- Updated config to include reference to the REST server details
- Added app.set to server/app.js that stores value from config file or if set the environment variable REST_SERVER_CONFIG
- Updated websocket in server/app.js to prevent app death on disconnection from remote and set it to use the config set so that it listens to the REST server for events
- Removed suspicious vehicles page
- Removed suspicious vehicles number from header
- Updated dashboard controller to work with v0.17.0 and remove composer events from VDA list e.g. AddUser.
- Updated daashboard controller to display the identities involved in the transaction correctly by name rather than using hard coded defaults
- Updated to use a single websocket to listen for events coming from the REST server

# Relies upon
- [ ] vehicle-manufacture-network being created as a sample in http://github.com/hyperledger/composer-sample-networks
- [ ] install will only function if docker image released using new vehicle-manufacture naming
- [ ] hlfv11 installer's node red flow relies on updated composer node red modules available <span>@</span>next

# PLEASE DO NOT MERGE UNTIL RELIED UPON TASKS ARE COMPLETE

### CODE REVIEW

**Comment by awjh-ibm** [comment]
~~Sample network appears on npm but version does not match 0.1.1 only at version 0.2.0~~

Seems as though there is a version at 0.1.14

File: packages/vehicle-manufacture-car-builder/config/default.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "restServer": {
+      "webSocketURL": "ws://localhost:3000",
+      "httpURL": "http://localhost:3000/api",
+      "accessToken": "DLHyHuJAbrzUwAAZ73Mh4B5D9zMxEVVpl0jOCEg8KHDddxkXHRMqTqg48bpjQHiT"
```
Comment (cazfletch) [inline packages/vehicle-manufacture-car-builder/config/default.json]: Why do we have an access token here?

File: packages/vehicle-manufacture-manufacturing/config/default.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "restServer": {
+      "webSocketURL": "ws://localhost:3000",
+      "httpURL": "http://localhost:3000/api",
+      "accessToken": "DLHyHuJAbrzUwAAZ73Mh4B5D9zMxEVVpl0jOCEg8KHDddxkXHRMqTqg48bpjQHiT"
```
Comment (cazfletch) [inline packages/vehicle-manufacture-manufacturing/config/default.json]: again with the access token

File: packages/vehicle-manufacture-vda/config/default.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "restServer": {
+      "webSocketURL": "ws://localhost:3000",
+      "httpURL": "http://localhost:3000/api",
+      "accessToken": "DLHyHuJAbrzUwAAZ73Mh4B5D9zMxEVVpl0jOCEg8KHDddxkXHRMqTqg48bpjQHiT"
```
Comment (cazfletch) [inline packages/vehicle-manufacture-vda/config/default.json]: access token???

**Review by cazfletch — CHANGES_REQUESTED** [review]

**Comment by awjh-ibm** [comment]
Hmm think they were a hangover from when I had a play with multiuser mode I'll work at taking them out. Also found that the VDA completely crashes if there are no transactions so I will work to fix that in the same commit

**Review by cazfletch — DISMISSED** [review]
LGTM

**Review by cazfletch — APPROVED** [review]

**Pull request merged**